### PR TITLE
rats-tls: support csv remote attestation

### PR DESCRIPTION
**Pull request by wojiaohanliyang**

### CODE REVIEW

File: src/attesters/csv/collect_evidence.c [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+static void *gva_to_gpa(void *va)
+{
+	int fd = -1;
+	void *pa = NULL;
+	uint64_t paging_entry_offset;
+	page_map_entry_t entry;
+
+	fd = open(PAGE_MAP_FILENAME, O_RDONLY);
+	if (fd == -1) {
+		RTLS_DEBUG("open %s fail\n", PAGE_MAP_FILENAME);
```
Comment (haosanzi) [inline src/attesters/csv/collect_evidence.c:42]: Please use `RTLS_ERR`

File: src/attesters/csv/collect_evidence.c [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+		return NULL;
+	}
+
+	paging_entry_offset = ((uint64_t)va >> PAGE_MAP_PAGE_SHIFT) * PAGE_MAP_ENTRY_SIZE;
+	if (lseek(fd, paging_entry_offset, SEEK_SET) == -1) {
+		RTLS_DEBUG("seek fail\n");
```
Comment (haosanzi) [inline src/attesters/csv/collect_evidence.c:48]: ditto

File: src/attesters/csv/csv_utils.c [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		return -1;
+
+	while (!feof(pipe))
+		fread(output, 1, sizeof(output), pipe);
+
+	pclose(pipe);
+	return 0;
+}
+
+int read_file(const char *filename, void *buffer, size_t len)
```
Comment (haosanzi) [inline src/attesters/csv/csv_utils.c:27]: The read_file is implemented in [link](https://github.com/inclavare-containers/rats-tls/blob/master/src/verifiers/sev/sev_utils.c#L12), please reuse this functiom

File: README.md [hunk 4]
```diff
@@ -65,6 +65,7 @@ Right now, RATS TLS supports the following instance types:
 | ----------- | --------------------- | -------------------------- | -------------------------- | ----------------------- |
 | 0         | nulltls               | nullattester               | nullverifier               | nullcrypto              |
 | 15        | openssl               | sgx\_la                    | sgx\_la                    | openssl                 |
+| 20        | openssl               | csv                        | csv                        | openssl                 |
 | 35        | openssl               | sev                        | sev                        | openssl                 |
 | 42        | openssl               | sev\_snp                   | sev\_snp                   | openssl                 |
 | 42        | openssl               | tdx\_ecdsa                 | tdx\_ecdsa                 | openssl                 |
```
Comment (haosanzi) [inline README.md:71]: Please add prerequisites about running CSV remote attestation.

File: src/verifiers/csv/csv_utils.c [hunk 5]
```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2022 Hygon Corporation
+ * Copyright (c) 2020-2022 Alibaba Cloud
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
```
Comment (haosanzi) [inline src/verifiers/csv/csv_utils.c:7]: Is `#include <stdio.h>` necessary?

Comment (wojiaohanliyang) [reply src/verifiers/csv/csv_utils.c:7]: It's not necessary, i will delete this line

File: README.md (hunk 4 above)
Comment (wojiaohanliyang) [reply README.md:71]: There are no special prerequisites for CSV remote attestation

File: src/verifiers/csv/verify_evidence.c [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+	if (sm2_verify_attestation_report(pek_cert, report) != 1) {
+		RTLS_ERR("failed to verify csv attestation report\n");
+		return err;
+	}
+
+	return ENCLAVE_VERIFIER_ERR_NONE;
+}
+
+
```
[the remark this answers is not in the document]
Comment (wojiaohanliyang) [reply src/verifiers/csv/verify_evidence.c:83]: I'll delte this empty line

File: src/attesters/csv/csv_utils.c (hunk 3 above)
Comment (wojiaohanliyang) [reply src/attesters/csv/csv_utils.c:27]: I'll touch a new file src/common/utils.c that contains function read_file

File: src/attesters/csv/csv_utils.c [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <stddef.h>
+#include <rats-tls/log.h>
+#include <openssl/evp.h>
+
+int execute_system_command(const char *cmdline)
```
[the remark this answers is not in the document]
Comment (wojiaohanliyang) [reply src/attesters/csv/csv_utils.c:12]: I'll replace it with system()

File: src/attesters/csv/collect_evidence.c (hunk 2 above)
Comment (wojiaohanliyang) [reply src/attesters/csv/collect_evidence.c:48]: I'll replace these RTLS_DEBUG with RTLS_ERR

File: src/attesters/csv/csv_utils.c (hunk 3 above)
Comment (wojiaohanliyang) [reply src/attesters/csv/csv_utils.c:27]: I'll insert '#include "../../verifiers/sev/sev_utils.c"' into src/attesters/csv/collect_evidence.c, and delete the definition of read_file in csv attester.  Test on these changes have been passed.

File: README.md (hunk 4 above)
Comment (wojiaohanliyang) [reply README.md:71]: I have explain how to get relevant patches in README.md

**Review by fenghao429 — APPROVED** [review]

**Review by fenghao429 — COMMENTED** [review]
csv attester instance is registered automatically

**Review by fenghao429 — APPROVED** [review]

**Review by fenghao429 — APPROVED** [review]

**Review by fenghao429 — APPROVED** [review]

**Review by fenghao429 — APPROVED** [review]

**Review by fenghao429 — APPROVED** [review]

**Review by haosanzi — APPROVED** [review]

**Pull request merged**